expressServer/scripts/encryptExistingUsers.js

```diff
@@ -1,38 +1,50 @@
 const mongoose = require('mongoose');
 const dotenv = require('dotenv');
-const app = require('express');
 const crypto = require('crypto');
+const { connectDB, closeDB } = require('../libraries/database.js');
+const { hashPassword } = require('../libraries/hashPassword.js');
 
 dotenv.config();
 
-mongoose.connect(process.env.MONGODB_URI, { useNewUrlParser: true, useUnifiedTopology: true })
-  .catch(err => console.error('Could not connect to MongoDB', err));
-
 const userSchema = new mongoose.Schema({
     username: { type: String, required: true, unique: true },
-    hash: { type: String, required: true },
+    hash: { type: String, required: false },
     salt: { type: String, required: false }
 });
 
 const User = mongoose.model('User', userSchema);
 
-function hashPassword(password, salt) {
-    return crypto.createHash('sha3-512').update(password + salt).digest('hex');
-}
-
 async function encryptAllUserPasswords() {
+    const batchSize = 1000;
+    let totalProcessed = 0;
+
     try {
-        const users = await User.find({});
-        for (let user of users) {
-            const salt = crypto.randomBytes(256).toString('hex');
+        await connectDB();
+
+        while (true) {
+            const users = await User.find({}).limit(batchSize);
```

Comment:
If you use limit strategy don't you need an order by and a offset. 

- In this case you pick the first 1000 wouldn't the second iteration pick the same 1000? 
- You are not sorting the query, in the databases there isn't order, so if you don't sort wouldn't you reprocess the same elements from other batches?